pkg/specgen/generate/pod_create.go

```diff
@@ -290,6 +291,19 @@ func PodConfigToSpec(rt *libpod.Runtime, spec *specgen.PodSpecGenerator, infraOp
 		}
 
 		spec.InfraContainerSpec = infraSpec
+		matching, err := json.Marshal(infraSpec)
```

Comment:
What does this code do, it looks like it marshals and unmarshals the same thing?

Reply:
until I rework the Pod API, this needs to be done in order to make the podspecgen a viable option. since infra holds a lot of the information for pods, the pod spec is kind of empty with overlapping fields. This maps the overlapping fields back to the pod spec since we are not allowed to expose infra's spec in json at the moment.